internal/controllers/machinedeployment/machinedeployment_status.go

```diff
@@ -82,11 +93,41 @@ func (r *Reconciler) updateStatus(ctx context.Context, s *scope) (retErr error)
 // as a consequence it is required to compute the counters again before calling scale down machine sets,
 // and again to before computing the overall availability of the Machine deployment.
 func setReplicas(machineDeployment *clusterv1.MachineDeployment, machineSets []*clusterv1.MachineSet) {
+	machineDeployment.Status.Replicas = mdutil.GetActualReplicaCountForMachineSets(machineSets)
 	machineDeployment.Status.ReadyReplicas = mdutil.GetReadyReplicaCountForMachineSets(machineSets)
 	machineDeployment.Status.AvailableReplicas = mdutil.GetAvailableReplicaCountForMachineSets(machineSets)
 	machineDeployment.Status.UpToDateReplicas = mdutil.GetUptoDateReplicaCountForMachineSets(machineSets)
 }
 
+func setPhase(_ context.Context, machineDeployment *clusterv1.MachineDeployment, machineSets []*clusterv1.MachineSet, getAndAdoptMachineSetsForDeploymentSucceeded bool) {
+	if !getAndAdoptMachineSetsForDeploymentSucceeded || machineDeployment.Spec.Replicas == nil {
+		machineDeployment.Status.Phase = string(clusterv1.MachineDeploymentPhaseUnknown)
+		return
+	}
+
+	desiredReplicas := ptr.Deref(machineDeployment.Spec.Replicas, 0)
```

Comment:
```suggestion
	desiredReplicas := *machineDeployment.Spec.Replicas
```
For clarity. The nil case is unreachable because of l.103